packages/core/src/utils/url.ts

```diff
@@ -10,59 +10,69 @@ import type { Dialect, DialectOptions, Options } from '../sequelize';
  * @param connectionString string value in format schema://username:password@host:port/database
  */
 export function parseConnectionString(connectionString: string): Options {
-  const urlParts = url.parse(connectionString, true);
   const options: Options = {};
-  if (urlParts.protocol) {
-    let protocol = urlParts.protocol.replace(/:$/, '');
+
+  // The following connectionStrings are not valid URLs, but they are supported by sqlite.
+  if (connectionString.toLowerCase() === 'sqlite://:memory:' || connectionString.toLowerCase() === 'sqlite::memory:') {
+    options.dialect = 'sqlite';
+    options.host = '';
+
+    return options;
+  }
+
+  const urlObject = new URL(connectionString);
+
+  if (urlObject.protocol) {
+    let protocol = urlObject.protocol.replace(/:$/, '');
     if (protocol === 'postgresql') {
       protocol = 'postgres';
     }
 
     options.dialect = protocol as Dialect;
```

Comment:
Do we already have something for that?

Reply:
I don't think we do, but the `Dialect` type can be transformed into this:

```
export const SUPPORTED_DIALECTS = ['mysql', 'postgres', 'sqlite', 'mariadb', 'mssql', 'mariadb', 'mssql', 'db2', 'snowflake', 'ibmi'] as const;

export type Dialect = typeof SUPPORTED_DIALECTS[number];
```

Reply:
That's a good point! Implemented that

Had some issues with defining the type in sequelize-typescript.ts and having it exported as part of `@sequelize/core` so I've just updated the Dialect type in sequelize.d.ts and added the SUPPORTED_DIALECTS to sequelize-typescript.ts

Also had to update the test that used `pg`. As per the [postgres docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING-URIS) it can be either `postgresql://` or `postgres://` so no need to add support for `pg://`